src/docs/arch-connection.dox

```diff
@@ -0,0 +1,59 @@
+/*! @arch_page arch-connection Connection
+
+@section arch_conn_def Definition
+
+A connection is a handle to a WiredTiger database instance. The connection has exclusive access to
+the database through a file lock, hence only one connection can be opened at a time. Internally, a
+connection is represented by \c WT_CONNECTION.
+
+@section arch_conn_lifecycle Life cycle
+
+@subsection arch_conn_init Initialization
+
+A connection is initialized when WT_CONNECTION::wiredtiger_open is called by the user application.
+WT_CONNECTION::wiredtiger_open accepts a list of configuration items (see @ref database_config) that
+can be used to enable different WiredTiger features and tune their behavior. Those features are for
+example related to @ref arch-eviction, @ref arch-logging, @ref arch-checkpoint, @ref arch-cache,
+statistics, etc. All the different available configuration settings are described in the
+documentation for WT_CONNECTION::wiredtiger_open.
+
+WT_CONNECTION::wiredtiger_open also performs different sanity checks depending on the configuration
+item "create". When "create" is specified and a database does not already exist, a new database is
+created along with specific WiredTiger files such as the turtle file and other metadata files. If a
+database already exists, whether "create" is specified or not, WiredTiger will try to open it and
+check for the existence of the different required WiredTiger files. If "create" is not specified,
+WiredTiger expects a previously created database where it is executed. If the existing database is
+corrupted and cannot be opened, either \c WT_RUN_RECOVERY or \c WT_TRY_SALVAGE error (see @ref
+error_handling) is returned to the user application and the connection is not created. In this case,
+a recovery operation will be required to bring the database to a consistent state (see @ref
+command_line for more details) before a connection can be successfully established with the
+database.
+
+Once the database has been successfully opened, internal worker threads are started to provide
+global services used at runtime. Those services consist of different threads to handle statistics,
+logging, eviction, checkpoint and cache management. The sweeping server that manages the active and
+inactive dhandles is started too, see @ref arch-dhandle for more information.
+
+Finally, before the connection is completely initialized, the database is set to a consistent state
+by running rollback to stable, see @ref arch-rts for more details.
+
+@subsection arch_conn_runtime Runtime
+
+At runtime, database-wide operations can be executed using the connection interface. For instance,
```

Comment:
This might be a bit of a nitpick (so completely feel free to ignore!), though for flow, I feel a good way to open up this section would be to highlight the relationship between a connection and a session first. This highlighting a key piece of functionality behind a connection  e.g 
```
To perform most CRUD data operations at runtime, a session must be opened on a WiredTiger connection. Through its lifetime, a connection will encapsulate multiple sessions...
```

Reply:
@alisonfel, thanks for the suggestion.
I like the current flow because we start from database-wide operations that can be triggered using the connection interface to "smaller" CRUD operations that require a session.
Would be great to know what the others think !
@bvpvamsikrishna @hbokhari @tammybailey

Reply:
I believe the current flow of information makes sense to me.